src/rdkafka_zstd.c

```diff
@@ -60,10 +60,7 @@ rd_kafka_resp_err_t rd_kafka_zstd_decompress(rd_kafka_broker_t *rkb,
                 break;
         }
 
-        /* Increase output buffer until it can fit the entire result,
-         * capped by message.max.bytes */
-        while (out_bufsize <=
-               (unsigned long long)rkb->rkb_rk->rk_conf.recv_max_msg_size) {
+        while (1) { /* will only loop if decompressed does not fit buffer */
```

Comment:
Yes indeed. What about using a different (maybe new) parameter for this? I think it will be better than reusing an existing parameter also used for different purposes. I can make the change and any naming suggestion is warmly welcome :)